### PR TITLE
added "note new orders with fulfillment locations" task

### PR DESCRIPTION
**Pull request by acflint**
Created a new task, based on the existing "Auto-tag new orders with their fulfillment locations" task, that adds a note to each new order with a comma separated list of the fulfillment locations.

### CODE REVIEW

**Comment by acflint** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by isaacbowen** [comment]
Hey @acflint! :) Thanks for this! It looks like the clabot is having some troubles; I'm going to get that sorted out so we can keep this pr moving.

**Comment by isaacbowen** [comment]
clabot has been mollified, and your name is now properly listed in the signatures file as of 2349c6c, despite the incorrect commit message. ;) I've pushed a minor whitespace fix in https://github.com/lightward/mechanic-tasks/pull/136/commits/7e9a571ee87d4cfdb7a7b65f771ffabea838d76b, too.

I do want to get one more set of reviewer eyes on this, so once @tekhaus signs off, we'll ship this! 🙌 Thanks a ton!

File: docs/note-new-orders-with-their-fulfillment-locations/script.liquid [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        }
+      }
+    }
+  {% endcapture %}
+
+  {% assign result = result_json | parse_json %}
+{% endif %}
+
+{% assign fulfillment_location_names = result.data.order.fulfillmentOrders.edges | map: "node" | map: "assignedLocation" | map: "name" | join: ", " %}
+{% assign note = "Fulfilled from: " | append: fulfillment_location_names %}
```
Comment (tekhaus) [inline docs/note-new-orders-with-their-fulfillment-locations/script.liquid:56]: There could be an existing order note, so you probably want to make sure and check for it and append this new info afterwards if one exists (after a newline or two).

File: docs/note-new-orders-with-their-fulfillment-locations/README.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+shopify/orders/create
+```
+
+[Learn about event subscriptions in Mechanic](https://learn.mechanic.dev/core/tasks/subscriptions)
+
+## Documentation
+
+Monitoring for brand-new orders, this task add a note to each order with the fulfillment locations that Shopify automatically assigns at the moment of order creation.
+
+Important: This task requires Shopify API version 2020-01 or later.
```
Comment (tekhaus) [inline docs/note-new-orders-with-their-fulfillment-locations/README.md:23]: I believe this note can be removed (I'm aware this is a vestige of the original task you based this new one on).

A task cannot be configured with an API version beyond the last supported Shopify API version (4 quarters prior) and Mechanic will auto bump task API versions when they are getting close to the unsupported date.

**Review by tekhaus — COMMENTED** [review]
Hey Alex, thanks a ton for contributing to the library! All looks good but for a couple of considerations.

File: docs/note-new-orders-with-their-fulfillment-locations/script.liquid (hunk 1 above)
Comment (acflint) [reply docs/note-new-orders-with-their-fulfillment-locations/script.liquid:56]: Thanks @tekhaus! A bit of a silly question, but how do I put new lines in the note string? When I do `{% assign note = result.data.order.note | append: "\n\nFulfilled from: " | append: fulfillment_location_names %}` it escapes the backslash and I end up with this: "Fulfilled from: PO Box 21122\n\nFulfilled from: PO Box 21122"

What am I doing wrong here? Thanks!

Comment (acflint) [reply docs/note-new-orders-with-their-fulfillment-locations/script.liquid:56]: Maybe I should use `capture` instead of `assign`. That seems to work okay.

Comment (acflint) [reply docs/note-new-orders-with-their-fulfillment-locations/script.liquid]: Using `capture` works, but requires some odd white spacing for the result to be clean. Let me know if you think there's a better way.

**Comment by tekhaus** [comment]
Hey Alex, Mechanic's version of Liquid conveniently has the _newline_
keyword literal to help out here.

https://learn.mechanic.dev/platform/liquid/keyword-literals/newline

**Review by tekhaus — APPROVED** [review]
Looks great!

**Comment by isaacbowen** [comment]
The task is live! :) Nice work, @acflint. 👏 And thanks for the review work, @tekhaus!

https://tasks.mechanic.dev/note-new-orders-with-their-fulfillment-locations

**Pull request merged**